docs/source/1.0/spec/core/model.rst

```diff
@@ -613,10 +614,11 @@ example is ``smithy.example#StringSet$member``.
 
 .. rubric:: Language support for sets
 
-Not all programming languages support set data structures. Such languages
-SHOULD represent sets as a custom set data structure that can interpret value
-hash codes and equality, or alternatively, store the values of a set data
-structure in a list and rely on validation to ensure uniqueness.
+Not all programming languages support an ordered set data structures. Such
+languages SHOULD represent sets as a custom set data structure that can
+interpret value hash codes and equality, or alternatively, store the values
```

Comment:
This is hinting at the fact that the backing data structure should be a `HashSet`. If that's the case I would call it out more explicitly; if it's not I would mention that an ordered set whose keys are not hashable is fine too.

Reply:
I don't think so. We should try to avoid that level of implementation detail showing up in the spec. This was the previous text, and I'll just remove it.